### PR TITLE
Unit Test 

### PR DESCRIPTION
**Pull request by MinJiang00**

### CODE REVIEW

File: dns_config/test/api_acl_test.go [hunk 1]
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
...
+func (f RoundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
 
-func Test_dns_config_AclAPIService(t *testing.T) {
+func NewTestClient(fn RoundTripFunc) *http.Client {
+	return &http.Client{
+		Transport: RoundTripFunc(fn),
+	}
+}
```
Comment (mathewab) [inline dns_config/test/api_acl_test.go]: can you move this to `internal/utils`?

File: dns_config/test/api_acl_test.go [hunk 2]
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
...
 
-	configuration := internal.NewConfiguration()
-	apiClient := openapiclient.NewAPIClient(configuration)
+func Test_dns_config_AclAPIService(t *testing.T) {
 
 	t.Run("Test AclAPIService AclCreate", func(t *testing.T) {
 
-		t.Skip("skip test") // remove to run test
+		//t.Skip("skip test") // remove to run test
```
Comment (mathewab) [inline dns_config/test/api_acl_test.go]: can you please remove commented code?

File: dns_config/test/api_auth_nsg_test.go [hunk 3]
```diff
@@ -1,93 +1,237 @@
 /*
 DNS Configuration API
 
-Testing AuthNsgAPIService
+Testing AclAPIService
```
Comment (mathewab) [inline dns_config/test/api_auth_nsg_test.go]: Shouldn't that be as it was - `AuthNsgAPIService`?

File: dns_config/test/api_acl_test.go [hunk 4]
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
...
+				Body:       io.NopCloser(bytes.NewReader(body)),
+				Header:     map[string][]string{"Content-Type": {"application/json"}},
+			}
+		})
+
+		configuration := internal.NewConfiguration()
+		configuration.HTTPClient = testClient
+
+		aclAPI := dns_config.NewAPIClient(configuration)
+		ctx := context.Background()
```
Comment (mathewab) [inline dns_config/test/api_acl_test.go]: you could have a common function for this in each package. 

```
func newAPITestClient(testFn RoundTripFunc) *dns_config.APIClient {
    configuration := internal.NewConfiguration()
    configuration.HTTPClient = NewTestClient(testFn)

   return dns_config.NewAPIClient(configuration)
}
```
That way you can avoid some duplication and simply write 

```
dnsConfigAPI := newAPITestClient(func(req *http.Request) *http.Response {
	require.Equal(t, "PATCH", req.Method)
        .....
}

File: dns_config/test/api_acl_test.go [hunk 5]
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
...
+			Comment: openapiclient.PtrString("This is a dummy ACL for testing."),
+			Id:      openapiclient.PtrString("dummyAclId"),
+			Name:    "dummyAclName",
+		}
 
-		require.Nil(t, err)
-		require.NotNil(t, resp)
-		assert.Equal(t, 200, httpRes.StatusCode)
+		testClient := NewTestClient(func(req *http.Request) *http.Response {
+			require.Equal(t, "POST", req.Method)
```
Comment (mathewab) [inline dns_config/test/api_acl_test.go]: use the constant from `http` package for all methods and status 

```suggestion
			require.Equal(t, http.MethodPost, req.Method)
```

File: dns_config/test/api_acl_test.go [hunk 6]
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
...
-	t.Run("Test AclAPIService AclDelete", func(t *testing.T) {
+			response := dns_config.ConfigCreateACLResponse{
+				Result: &dummyAcl,
+			}
+			body, err := json.Marshal(response)
+			require.NoError(t, err)
 
-		t.Skip("skip test") // remove to run test
+			return &http.Response{
+				StatusCode: 200,
```
Comment (mathewab) [inline dns_config/test/api_acl_test.go]:
```suggestion
				StatusCode: http.StatusOk,
```

File: dns_config/test/api_acl_test.go [hunk 7]
```diff
@@ -7,87 +7,244 @@ Testing AclAPIService
...
 
-		t.Skip("skip test") // remove to run test
+		//t.Skip("skip test") // remove to run test
 
-		resp, httpRes, err := apiClient.AclAPI.AclCreate(context.Background()).Execute()
+		dummyAcl := dns_config.ConfigACL{
+			Comment: openapiclient.PtrString("This is a dummy ACL for testing."),
+			Id:      openapiclient.PtrString("dummyAclId"),
+			Name:    "dummyAclName",
+		}
```
Comment (mathewab) [inline dns_config/test/api_acl_test.go]: Put all object definitions that are reused outside the Test function.

File: keys/api_tsig.go [hunk 8]
```diff
@@ -13,6 +13,7 @@ package keys
 import (
 	"bytes"
 	"context"
+	_ "github.com/infobloxopen/bloxone-go-client/dns_config"
```
Comment (mathewab) [inline keys/api_tsig.go]: why this change?

File: keys/model_keys_tsig_key.go [hunk 9]
```diff
@@ -38,6 +39,7 @@ type KeysTSIGKey struct {
 	Tags map[string]interface{} `json:"tags,omitempty"`
 	// Time when the object has been updated. Equals to _created_at_ if not updated after creation.
 	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+	res       interface{}
```
Comment (mathewab) [inline keys/model_keys_tsig_key.go]: why did you add this?

File: keys/.openapi-generator/VERSION [hunk 10]
```diff
@@ -1 +1 @@
-7.0.1
+7.2.0
```
Comment (venkat-iblox) [inline keys/.openapi-generator/VERSION]: let's stick to using generator 7.0.1 for now

File: dns_config/test/api_acl_test.go [hunk 11]
```diff
@@ -7,87 +7,158 @@ Testing AclAPIService
...
 	"github.com/infobloxopen/bloxone-go-client/internal"
 )
 
-func Test_dns_config_AclAPIService(t *testing.T) {
-
-	configuration := internal.NewConfiguration()
-	apiClient := openapiclient.NewAPIClient(configuration)
-
-	t.Run("Test AclAPIService AclCreate", func(t *testing.T) {
+var dns_config_Post = openapiclient.ConfigACL{
```
Comment (mathewab) [inline dns_config/test/api_acl_test.go]: variable names should be camelCase

File: dns_config/test/api_acl_test.go [hunk 12]
```diff
@@ -7,87 +7,158 @@ Testing AclAPIService
...
-		t.Skip("skip test") // remove to run test
+var dns_config_Patch = openapiclient.ConfigACL{
+	Comment: openapiclient.PtrString("This is an updated dummy ACL for testing."),
+	Id:      dns_config_Post.Id,
+	Name:    "updatedDummyAclName",
+	Tags:    make(map[string]interface{}),
+}
 
-		resp, httpRes, err := apiClient.AclAPI.AclCreate(context.Background()).Execute()
+func Test_dns_config_AclAPIService(t *testing.T) {
```
Comment (mathewab) [inline dns_config/test/api_acl_test.go]: You can remove the package name from the tests

```suggestion
func TestAclAPIService(t *testing.T) {
```

File: dns_config/test/api_auth_zone_test.go [hunk 13]
```diff
@@ -5,101 +5,187 @@ Testing AuthZoneAPIService
...
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	openapiclient "github.com/infobloxopen/bloxone-go-client/dns_config"
 	"github.com/infobloxopen/bloxone-go-client/internal"
 )
 
-func Test_dns_config_AuthZoneAPIService(t *testing.T) {
+var ConfigCopyAuthZonePost = openapiclient.ConfigCopyAuthZone{
+	Comment: openapiclient.PtrString("This is a copyping AuthZone for testing."),
```
Comment (mathewab) [inline dns_config/test/api_auth_zone_test.go]: `copyping` typo?

File: ipam/test/api_dhcp_host_test.go [hunk 14]
```diff
@@ -7,76 +7,127 @@ Testing DhcpHostAPIService
 
 // Code generated by OpenAPI Generator (https://openapi-generator.tech);
```
Comment (mathewab) [inline ipam/test/api_dhcp_host_test.go]: This line is not removed in `ipam`

**Review by mathewab — APPROVED** [review]
